### PR TITLE
ATF test script for reproducing defect #1385.

### PR DESCRIPTION
**Pull request by mkorniichuk**
ATF Test Scripts to check [#1385](https://github.com/smartdevicelink/sdl_core/issues/1385)

This PR is **ready** for review.

### Summary
Script for reproducing defect.

### ATF version
[develop](https://github.com/smartdevicelink/sdl_atf/tree/develop)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+-- 1) SDL receives BasicCommunication.OnEventChanged(Phone_Call,true) from HMI when apps are full.
+-- Expected:
+-- SDL should send to mobile app OnHMIStatus
+--   1) For navi app: hmiLevel = "LIMITED", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"
+--   2) For media app: hmiLevel = "BACKGROUND", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"
+--   3) For voice communication app: hmiLevel = "BACKGROUND", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"
+---------------------------------------------------------------------------------------------------
+
+-- [[ Required Shared libraries ]]
+local config = require("local_config")
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua]: config is global variables, so can be not defined.

File: test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    config.application1.registerAppInterfaceParams[key] = value
+  end
+end
+
+local function sendOnEventChanged(testCase)
+  common.getHMIConnection():SendNotification("BasicCommunication.OnEventChanged", {
+    eventName = "PHONE_CALL",
+    isActive = true })
+  common.getMobileSession():ExpectNotification("OnHMIStatus", cases[testCase][expectedHmiStatus])
+  common.getHMIConnection():SendNotification("BasicCommunication.OnEventChanged", {
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua]: isActive = false must be send in block Do by receiving ferst OnHMIStatus notification, 
After second OnEventChanged sdl must restore previous HMI level, so need to expect 2 OnHMINotifications

File: test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    connect.mobileSession[i] = nil
+  end
+end
+
+local function unregisterApp()
+  local cid = common.getMobileSession():SendRPC("UnregisterAppInterface", {})
+  common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered",
+  { appID = common.getHMIAppId(), unexpectedDisconnect = false })
+  common.getMobileSession():ExpectResponse(cid, {success = true, resultCode = "SUCCESS"})
+  cleanSessions()
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua]: cleanSessions must be executed after receiving UnregisterAppInterface response and aslo before clean the session must be stoped. 
step with session stop can be added to cleanSessions like `connect.mobileSession[i]:Stop()`

File: test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+
+for key, value in pairs(cases) do
+  runner.Title("TestCase: " .. key)
+  runner.Step("Prepare application config", prepareConfig, { value.params })
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua]: better Prepare application params

File: test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+
+for key, value in pairs(cases) do
+  runner.Title("TestCase: " .. key)
+  runner.Step("Prepare application params", prepareConfig, { value.params })
+  runner.Step("Register App1", common.registerAppWOPTU, { 1 })
+  runner.Step("Activate App1", common.activateApp, { 1 })
+
+  runner.Step("On event changed", sendOnEventChanged, { key })
```
Comment (ShobhitAd) [inline test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua]: `sendOnEventChanged, { key }` causes error `PANIC: unprotected error in call to Lua API (./modules/atf/util.lua:124: bad argument #1 to 'pairs' (table expected, got string))`

Comment (mkorniichuk) [reply test_scripts/Defects/5_0/1385_SDL_doesnt_deactivate_app_when_receives_OnEventChange_PHONE_CALL_from_HMI.lua]: Fixed in sendOnEventChanged function. Thanks for the feedback.

**Comment by dboltovskyi** [comment]
@ShobhitAd please do not review this PR for now since it hasn't passed our internal review yet.  
Review and appropriate updates going be done till 1st of November.
Thank you.

**Comment by dboltovskyi** [comment]
@ShobhitAd Please notice this PR is ready for review.

**Comment by GetmanetsIrina** [comment]
PR is rebased on current develop(a2040db) branch.
The script is applicable to all policy flows and transport types.
Script has been moved to 6_1 folder.
The test set for the script is added.

**Comment by dboltovskyi** [comment]
Closed in a favor https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2543

**Pull request closed without merging**